components/webcompat/content/browser/webcompat_exceptions_service.cc

```diff
@@ -180,6 +144,46 @@ void WebcompatExceptionsService::OnJsonFileDataReady(
   }
 }
 
+PatternsByWebcompatTypeMap ReadAndParseJsonRules(
+    const base::FilePath& txt_file_path) {
+  PatternsByWebcompatTypeMap patterns_by_webcompat_type;
+  const auto raw_contents =
+      brave_component_updater::GetDATFileAsString(txt_file_path);
+  ParseJsonRules(patterns_by_webcompat_type, raw_contents);
+  return patterns_by_webcompat_type;
+}
+
+}  // namespace
+
+WebcompatExceptionsService::WebcompatExceptionsService(
+    LocalDataFilesService* local_data_files_service)
+    : LocalDataFilesObserver(local_data_files_service) {}
+
+void WebcompatExceptionsService::LoadWebcompatExceptions(
+    const base::FilePath& install_dir) {
+  base::FilePath txt_file_path =
+      install_dir.AppendASCII(WEBCOMPAT_EXCEPTIONS_JSON_FILE_VERSION)
+          .AppendASCII(WEBCOMPAT_EXCEPTIONS_JSON_FILE);
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE, {base::MayBlock()},
+      base::BindOnce(&ReadAndParseJsonRules, txt_file_path),
+      base::BindOnce(&WebcompatExceptionsService::SetRules,
+                     weak_factory_.GetWeakPtr()));
+}
+
+const std::vector<ContentSettingsPattern>
+WebcompatExceptionsService::GetPatterns(ContentSettingsType webcompat_type) {
+  base::AutoLock lock(lock_);
+  const auto it = patterns_by_webcompat_type_.find(webcompat_type);
+  return it == patterns_by_webcompat_type_.end() ? kEmptyPatternVector
+                                                 : it->second;
+}
+
+void WebcompatExceptionsService::SetRules(const PatternsByWebcompatTypeMap& patterns_by_webcompat_type) {
```

Comment:
this should be `PatternsByWebcompatTypeMap patterns_by_webcompat_type`. you can't move `const` types.

Reply:
fixed